source/gameengine/Ketsji/KX_TextureProbe.cpp

```diff
@@ -0,0 +1,250 @@
+/*
+* ***** BEGIN GPL LICENSE BLOCK *****
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public License
+* as published by the Free Software Foundation; either version 2
+* of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the Free Software Foundation,
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*
+* Contributor(s): Ulysse Martin, Tristan Porteries.
+*
+* ***** END GPL LICENSE BLOCK *****
+*/
+
+/** \file KX_TextureProbe.cpp
+ *  \ingroup ketsji
+ */
+
+#include "KX_TextureProbe.h"
+#include "KX_GameObject.h"
+#include "KX_Globals.h"
+
+#include "DNA_texture_types.h"
+
+KX_TextureProbe::KX_TextureProbe(EnvMap *env, KX_GameObject *viewpoint)
+	:m_viewpointObject(viewpoint),
+	m_invalidProjection(true),
+	m_enabled(true),
+	m_ignoreLayers(env->notlay),
+	m_clipStart(env->clipsta),
+	m_clipEnd(env->clipend),
+	m_lodDistanceFactor(1.0),
+	m_autoUpdate(true),
+	m_forceUpdate(true)
+{
+	m_autoUpdate = (env->flag & ENVMAP_AUTO_UPDATE) != 0;
+}
+
+KX_TextureProbe::~KX_TextureProbe()
+{
+}
+
+std::string KX_TextureProbe::GetName()
+{
+	return "KX_TextureProbe";
+}
+
+KX_GameObject *KX_TextureProbe::GetViewpointObject() const
+{
+	return m_viewpointObject;
+}
+
+void KX_TextureProbe::SetViewpointObject(KX_GameObject *gameobj)
```

Comment:
I'm not sure this should be allowed for mirrors (The viewpoint object must not be necessarly the mirror object itself? If we change viewpoint object, doesn't it make the mirror reflection inaccurate?). But I didn't thought enough about it to see in what case it could be useful for mirrors.